README.md

```diff
@@ -519,6 +521,11 @@ These settings can be changed at any time.
   Bottlerocket enables the `json-file`, `awslogs`, and `none` drivers by default.
 * `settings.ecs.loglevel`: The level of verbosity for the ECS agent's logs.
   Supported values are `debug`, `info`, `warn`, `error`, and `crit`, and the default is `info`.
+* `settings.ecs.metadata-service-rps`: The steady state rate limit of the throttling configurations set for the task metadata service
+* `settings.ecs.metadata-service-burst`: The burst rate limit of the throttling configurations set for the task metadata service
+* `settings.ecs.reserved-memory`: The amount of memory, in MB, reserved for critical system processes
+* `settings.ecs.task-cleanup-wait-duration`: Time to wait before the tasks resources are removed after they are stopped.
```

Comment:
```
ECS_ENGINE_TASK_CLEANUP_WAIT_DURATION
Time to wait from when a task is stopped until the Docker container is removed.

ECS_CONTAINER_STOP_TIMEOUT
Time to wait from when a task is stopped before its containers are forcefully stopped
if they do not exit normally on their own.
```

What happens if task-cleanup-wait is shorter than container-stop-timeout ? Or does the cleanup timer not start running until the last container stops? The docs imply that both timeouts run concurrently, starting from the instant the task is stopped.

Reply:
I addressed this offline, but here are my findings:

`ContainerStopTimeout` is used when the ECS agent signals docker that it should stop a container. After this call, the ECS agent updates the state of the container to `ContainerStopped`. Once all the task's containers are `ContainerStopped` , the task transitions to `TaskStopped`. There is an infinite loop that checks if the task is in `TaskStopped`, if it is, the agent will begin the cleanup after `TaskCleanupDuration` have passed.

Thus, even if `ContainerStopTimeout` is lower than `TaskCleanupDuration` , no race conditions will occur since the cleanup will only happen after all the containers have stopped.